### PR TITLE
PhysicsTools/KinFitter: fix gcc5 strict-overflow warning.

### PR DESCRIPTION
**Pull request by gartung**
Move conditional out of for statement so that compiler does not attempt to optimize out for loop.

This fixes the gcc5 warning:

  /build/cmsbld/jenkins-workarea/workspace/build-any-ib/w/tmp/BUILDROOT/7339d4514f5638002b3a1b1f515c3f0b/opt/cmssw/slc6_amd64_gcc530/cms/cmssw/CMSSW_9_1_X_2017-03-16-1100/src/PhysicsTools/KinFitter/src/TKinFitter.cc:1256:6: warning: assuming signed overflow does not occur when assuming that (X + c) < X is always false [-Wstrict-overflow]
  void TKinFitter::printMatrix(const TMatrixD &matrix, const TString& name) {
      ^

### CODE REVIEW

**Comment by gartung** [comment]
@Dr15Jones test please

**Comment by Dr15Jones** [comment]
Please add `{}` to aid in understanding the structure.

**Comment by gartung** [comment]
Done

File: PhysicsTools/KinFitter/src/TKinFitter.cc [hunk 1]
```diff
@@ -1282,18 +1282,18 @@ void TKinFitter::printMatrix(const TMatrixD &matrix, const TString& name) {
...
-      if (iCol < iSheet+colsPerSheet) 
-      log << std::setw(8) << iCol << "    |";
-    log << "\n"
-	<< topbar << " \n";
+	 << "     |";
+    for (Int_t iCol = iSheet; iCol < nCols; iCol++) {
+      if (iCol < iSheet+colsPerSheet) log << std::setw(8) << iCol << "    |";
+      log << "\n"
+	   << topbar << " \n";
+      }
```
Comment (Dr15Jones) [inline PhysicsTools/KinFitter/src/TKinFitter.cc]: This brace does not appear to line up with the `for`.

**Comment by gartung** [comment]
Done

File: PhysicsTools/KinFitter/src/TKinFitter.cc [hunk 2]
```diff
@@ -1282,16 +1282,18 @@ void TKinFitter::printMatrix(const TMatrixD &matrix, const TString& name) {
...
+	   << topbar << " \n";
+    }
     for(Int_t iRow = 0; iRow < nRows; iRow++) {
       log << std::setw(4) << iRow << " |";
-      for (Int_t iCol = iSheet; iCol < iSheet+colsPerSheet && iCol < nCols; iCol++)
-	log << std::setw(12) << matrix(iRow, iCol) << " ";
-      log << "\n";
+      for (Int_t iCol = iSheet; iCol < nCols; iCol++) {
+          if (iCol < iSheet+colsPerSheet ) log << std::setw(12) << matrix(iRow, iCol) << " ";
+          log << "\n";
```
Comment (Dr15Jones) [inline PhysicsTools/KinFitter/src/TKinFitter.cc]: This should be outside the for loop to match the previous logic.

File: PhysicsTools/KinFitter/src/TKinFitter.cc [hunk 3]
```diff
@@ -1282,16 +1282,18 @@ void TKinFitter::printMatrix(const TMatrixD &matrix, const TString& name) {
...
     log << "\n"
-	<< "     |";
-    for (Int_t iCol = iSheet; iCol < iSheet+colsPerSheet && iCol < nCols; iCol++)
-      log << std::setw(8) << iCol << "    |";
-    log << "\n"
-	<< topbar << " \n";
+	 << "     |";
+    for (Int_t iCol = iSheet; iCol < nCols; iCol++) {
+      if (iCol < iSheet+colsPerSheet) log << std::setw(8) << iCol << "    |";
+      log << "\n"
```
Comment (Dr15Jones) [inline PhysicsTools/KinFitter/src/TKinFitter.cc]: The curly brace for the `for` loop should be before this line to match the previous logic.

**Review by Dr15Jones — CHANGES_REQUESTED** [review]

**Comment by gartung** [comment]
I think I have it right this time.

**Comment by Dr15Jones** [comment]
please test

**Pull request merged**